console/frontend/src/main/frontend/src/app/services/poller.service.ts

```diff
@@ -23,243 +25,175 @@ export class PollerService {
-    return this.poller ? true : false;
+  start(): void {
+    if (this.state !== 'STOPPED') return;
+
+    this.runningSubscription = null;
+    this.windowIntervalId = window.setInterval(
+      () => this.run(),
+      this.intervalTime,
+    );
+    this.state = 'WAITING';
   }
 
   stop(): void {
-    if (!this.started()) {
-      this.waiting = true;
-      return;
-    }
+    if (this.state === 'STOPPED') return;
 
-    this.ai.list = [];
-    this.ai.avg = 0;
-    if (this.waiting) clearTimeout(this.poller);
-    else clearInterval(this.poller);
-    this.waiting = true;
-    delete this.poller;
+    if (this.state === 'RUNNING') this.runningSubscription?.unsubscribe();
+    window.clearInterval(this.windowIntervalId!);
+    this.windowIntervalId = null;
+    this.state = 'STOPPED';
   }
 
-  fn(runOnce: boolean = false): void {
-    this.fired++;
-    this.http
-      .get<unknown>(this.appService.absoluteApiPath + this.uri)
-      .subscribe({
-        next: this.callback,
-        error: () => {
-          this.addError();
-
-          let errors = 0;
-          for (const x in this.errorList) {
-            const y = this.errorList[x];
-            if (
-              this.fired == y.fired ||
-              this.fired - 1 == y.fired ||
-              this.fired - 2 == y.fired
-            )
-              errors++;
-          }
-          this.Debug.info(
-            `Encountered unhandled exception, poller[${this.uri}] eventId[${this.fired}] retries[${errors}]`,
-          );
-          if (errors < 3) return;
-
-          this.Debug.warn(
-            `Max retries reached. Stopping poller [${this.uri}]`,
-            this,
-          );
-
-          runOnce = true;
-          this.stop();
-        },
-        complete: () => {
-          if (runOnce) return;
-
-          if (this.waiting) this.start();
-        },
-      });
+  restart(): void {
+    this.stop();
+    this.start();
   }
 
-  run(): void {
-    this.fn(true);
+  runOnce(): void {
+    this.run(true);
   }
 
-  start(): void {
-    if (this.started() && !this.waiting) return;
+  private run(once?: boolean): void {
+    if (this.state === 'RUNNING') return;
 
-    if (this.waiting) {
-      const now = Date.now();
-      if (this.lastPolled) {
-        const timeBetweenLastPolledAndNow = now - this.lastPolled;
-        const interval = this.ai.push(timeBetweenLastPolledAndNow);
-        if (interval! > 0 && interval! > this.pollerInterval) {
-          this.setInterval(interval!, false);
-          this.waitForResponse(false);
-          return;
-        }
-      }
-      this.poller = window.setTimeout(() => this.fn(), this.pollerInterval);
-      this.lastPolled = now;
-    } else
-      this.poller = window.setInterval(() => this.fn(), this.pollerInterval);
-  }
+    this.errorCount = 0;
+    this.runningSubscription = this.http.get<T>(this.url).subscribe({
+      next: this.callback,
+      error: (error) => {
+        this.errorCount += 1;
```

Comment:
Maybe?

```suggestion
        this.errorCount++;
```

Reply:
I dont think there is any difference to which one you use but in my opinion is `+=` much clearer than using `var++`. 
We're not writing in C/C++ and this looks more explicit. Also then we dont have to argue about using `++var` and if thats more appropriate compared to `var++` (yes there technically is a difference)